creme/linear_model/adpredictor.py

```diff
@@ -0,0 +1,175 @@
+import collections
+import numbers
+import numpy as np 
+from scipy import stats
+from .. import base 
+from .. import optim
+from .. import utils
+
+
+
+__all__ = [
+    'AdPredictor'
+]
+
+
+class NormalPrior(optim.initializers.Initializer):
+    """Initializer used to set beliefs over weights of a model.
+
+   Initializer which represents a gaussian belief over a weight i.e. a normal prior with parameters μ ``mean`` and σ2 ``variance``. 
+   The ``mean`` is calculated taking the predictive posterior closed form inverse of the prior probability value.
+
+    Parameters:
+        prior_probability (float): Prior probability on the feature weights prior_probability = P(y=1 | x, weights).
+        beta (float): Scale of the inverse link function (standardized cumulative normal distribution function).
+        n_features (int): Number of features.
+
+    Example:
+
+        ::
+
+            >>> from creme import linear_model
+            >>> from creme import datasets
+
+            >>> X_y = datasets.Phishing()
+
+            >>> init = linear_model.adpredictor.NormalPrior(beta=0.05, n_features=X_y.n_features, prior_probability=0.6)
+
+            >>> print(init)
+            Biased normal prior on weights 𝒩(μ=2.281, σ2=1.000)
+
+
+    Attributes:
+        mean (float): Mean of the gaussian belief over the weight.
+        variance (float): Variance of the gaussian belief over the weight.
+
+    """
+    def __init__(self, beta, n_features, prior_probability=None):
+        self.beta = beta
+        self.n_features = n_features
+        self.prior_probability = prior_probability
+        self.mean = 0 if prior_probability is None else stats.norm.ppf(prior_probability) * (beta ** 2 + n_features)
+        self.variance = 1.0
+
+    def __str__(self):
+        return f'Unbiased normal prior on weights 𝒩(μ={self.mean:.3f}, σ2={self.variance:.3f})' if self.prior_probability is None else f'Biased normal prior on weights 𝒩(μ={self.mean:.3f}, σ2={self.variance:.3f})'
+
+    def __call__(self):
+        return self
+
+
+class BOPR:
```

Comment:
Is there any reason why this is it's own particular class? If we're not going to reuse then we may as well incorporate it's logic in the `AdPredictor` class.